src/texting/xlsx-builder.js

```diff
@@ -0,0 +1,116 @@
+const excel = require('excel4node');
+const Helper = require('../helper');
+
+class ExcelBuilder {
+    DATE_COLUMN_INDEX = 1; DATE_COLUMN_WIDTH = 20;
+    STATUS_COLUMN_INDEX = 2; STATUS_COLUMN_WIDTH = 20;
+    START_COLUMN_INDEX = 3; START_COLUMN_WIDTH = 15;
+    END_COLUMN_INDEX = 4; END_COLUMN_WIDTH = 15;
+    TOTAL_HOURS_COLUMN_INDEX = 5; TOTAL_HOURS_COLUMN_WIDTH = 25;
+    TASKS_COLUMN_INDEX = 6; TASKS_COLUMN_WIDTH = 50;
+
+    headerStyle;
+    statusOffStyle;
+    statusWorkingStyle;
+
+    buildReportFile(myMap, timeStart, timeEnd, uploadFunction, nameFile = `excel_report_${Math.floor(Math.random() * 1000)}`){
+        const wb = new excel.Workbook({
+            dateFormat: 'dd/mm/yyyy'
+        });
+        const ws = wb.addWorksheet('Hours report');
+
+        this.setWidth(ws);
+        this.createStyles(wb)
+        this.buildHeader(ws, this.headerStyle);
+        let row = 2;
+        for (let day = timeStart; day <= timeEnd; day.setDate(day.getDate() + 1)) {
+            let log = myMap.get(new Date(day.toDateString()))
+            this.buildRow(ws, day, row++, log);
+        }
+
+        const fileWithExtension = `${nameFile}.xlsx`;
+        wb.write(fileWithExtension, () => {
+            uploadFunction(fileWithExtension);
+        });
+    }
+
+    buildRow(ws, date, row, log){
+        const isWeekend = Helper.isWeekend(date)
+        ws.cell(row, this.DATE_COLUMN_INDEX)
+            .date(date);
+        ws.cell(row, this.STATUS_COLUMN_INDEX)
+            .string(isWeekend? 'Off' : 'Working').style(isWeekend? this.statusOffStyle: this.statusWorkingStyle);
+        ws.cell(row, this.START_COLUMN_INDEX)
+            .string('8:00');
+        ws.cell(row, this.END_COLUMN_INDEX)
+            .string('16:00');
```

Comment:
Not sure what to do with that, I don't think there is nice way to get this information anywhere